File: 392-is-subsequence/solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        s_len = len(s)
        t_len = len(t)
        if s_len == 0:
            return True
        if t_len < s_len:
            return False
        s_idx = 0
        for t_idx, i in enumerate(t):
            if s_idx < s_len and i == s[s_idx]:
                s_idx += 1
        return s_idx == s_len
        

File: 392-is-subsequence/solution.go
package main

func isSubsequence(s string, t string) bool {
	slen := len(s)
	tlen := len(t)
	if slen == 0 {
		return true
	}
	if tlen < slen {
		return false
	}
	sidx := 0
	for _, i := range t {
		if sidx < slen && i == rune(s[sidx]) {
			sidx++
		}
	}
	return sidx == slen
}
